File: leetcode/0189-rotate-array/solution.py
class Solution(object):
    def rotate(self, nums, k):
        n=len(nums)
        final=[0]*n
        for i in range(0,n):
            if (i + k)>=n:
                final[(i+k)%n]=nums[i]
            else:
                final[i+k]=nums[i]
        for i in range(n):
            nums[i]=final[i]
        return nums

        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        

File: leetcode/0189-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k %= n; 
        if (k == 0) return; 

        reverse(nums.begin(), nums.end());          // Reverse entire array
        reverse(nums.begin(), nums.begin() + k);    // Reverse first k
        reverse(nums.begin() + k, nums.end());
    }
};
